Sidecar task fails to start with interpolation in service stanza

### Nomad version
nomad v1.0.4

### Issue
When trying to start a consul connect job with service port interpolation, sidecar task fails to start . Nomad client has configuration:

```
{
  "client": {
  ...
    "node_class": "frame-server",
    "host_network": {
      "public": {
        "cidr": "<public-ip-addr>/32"
      },
      "private": {
        "cidr": "<private-ip-addr>/24"
      }
    },
    "meta": {
     ...
      "network": "lb_public",
    }
  }
}
```

### Job file (if appropriate)

```

```

```
job "prometheus_lb" {
  datacenters = ["euc1", "fsn1"]
  type        = "system"

  constraint {
    attribute = "${node.class}"
    operator  = "regexp"
    value     = "(frame-server|cloud-cpu-worker)+"
  }

  group "prometheus" {

    network {
      mode = "bridge"

      port "lb_public" {
        static       = 9090
        to           = 9090
        host_network = "public"
      }

      port "sidecar" {
        host_network = "private"
      }

      port "lb_private" {
        static       = 9090
        to           = 9090
        host_network = "private"
      }
    }

    service {
      name = "prometheus-https"
      port = "${meta.network}"

      connect {
        sidecar_service {
          port = "sidecar"
          proxy {
            expose {
              path {
                path            = "/metrics"
                protocol        = "http"
                local_path_port = 9102
                listener_port   = "stats"
              }
            }
            upstreams {
              destination_name = "prometheus"
              local_bind_port  = 9091
            }
          }
        }
      }
        
      check {
        type     = "http"
        path     = "/ping"
        interval = "5s"
        timeout  = "2s"

        check_restart {
          limit = 3
          grace = "30s"
          ignore_warnings = false
        }
      }
    }

    task "lb" {
      driver = "docker"

      config {
        image      = "traefik:v2.4"
        force_pull = true

        args = [
          "--providers.file.filename=/local/traefik.yml",
          "--entryPoints.https.address=:9090",
          "--global.checknewversion=true",
          "--global.sendAnonymousUsage=true",
          "--log.level=WARN",
          "--accesslog=true",
          "--ping.entryPoint=https"
        ]
      }

      resources {
        cpu    = 100
        memory = 128
      }

      template {
        data = replace(yamlencode({
          tls = {
            options = {
              prometheus = {
                clientAuth = {
                  caFiles = ["/local/certs/prometheus-ca.crt"]
                  clientAuthType = "RequireAndVerifyClientCert"
                }
              }
            }
          }
          http = {
            routers = {
              prometheus = {
                entryPoints = ["https"]
                rule = "Host(`prometheus.{{env \"node.datacenter\"}}.{{env \"meta.environment\"}}.example.com`)"
                service = "prometheus"
                tls = {
                  options = "prometheus"
                }
              }
            }
            services = {
              prometheus = {
                loadBalancer = {
                  servers = [{
                    url = "http://{{env \"NOMAD_UPSTREAM_ADDR_prometheus\"}}/"
                  }]
                }
              }
            }
          }
        }), "\\", "")
        destination   = "/local/traefik.yml"
        change_mode   = "signal"
        change_signal = "SIGHUP"
      }

      template {
        data = <<EOH
{{ key "secrets/prometheus/tls/ca.crt" }}
EOH
        destination = "/local/certs/prometheus-ca.crt"
        change_mode   = "signal"
        change_signal = "SIGHUP"
      }
    }
  }
}


```

### Nomad Client logs (if appropriate)

```
2021-03-09T08:12:36.011+0100 [ERROR] client.alloc_runner.task_runner.task_hook.envoy_bootstrap: error creating bootstrap configuration for Connect proxy sidecar: alloc_id=b9cc3b58-9955-935a-214c-4030188d182f task=connect-proxy-prometheus-https error="exit status 1" stderr="==> No sidecar proxy registered for _nomad-task-b9cc3b58-9955-935a-214c-4030188d182f-group-prometheus-prometheus-https-${meta.network}
2021-03-09T08:12:13.862+0100 [INFO]  client.alloc_runner.task_runner: restarting task: alloc_id=b9cc3b58-9955-935a-214c-4030188d182f task=connect-proxy-prometheus-https reason="Restart within policy" delay=15.711183607s
```

--- a/client/allocrunner/taskrunner/envoy_bootstrap_hook.go
+++ b/client/allocrunner/taskrunner/envoy_bootstrap_hook.go
@@ -13,6 +13,7 @@ import (
 	"github.com/hashicorp/go-hclog"
 	"github.com/hashicorp/nomad/client/allocdir"
 	ifs "github.com/hashicorp/nomad/client/allocrunner/interfaces"
+	"github.com/hashicorp/nomad/client/taskenv"
 	agentconsul "github.com/hashicorp/nomad/command/agent/consul"
 	"github.com/hashicorp/nomad/helper"
 	"github.com/hashicorp/nomad/nomad/structs"
@@ -130,11 +131,12 @@ func (_ *envoyBootstrapHook) extractNameAndKind(kind structs.TaskKind) (string, 
 	return serviceKind, serviceName, nil
 }
 
-func (h *envoyBootstrapHook) lookupService(svcKind, svcName, tgName string) (*structs.Service, error) {
+func (h *envoyBootstrapHook) lookupService(svcKind, svcName, tgName string, taskEnv *taskenv.TaskEnv) (*structs.Service, error) {
 	tg := h.alloc.Job.LookupTaskGroup(h.alloc.TaskGroup)
+	interpolatedServices := taskenv.InterpolateServices(taskEnv, tg.Services)
 
 	var service *structs.Service
-	for _, s := range tg.Services {
+	for _, s := range interpolatedServices {
 		if s.Name == svcName {
 			service = s
 			break
@@ -167,7 +169,7 @@ func (h *envoyBootstrapHook) Prestart(ctx context.Context, req *ifs.TaskPrestart
 		return err
 	}
 
-	service, err := h.lookupService(serviceKind, serviceName, h.alloc.TaskGroup)
+	service, err := h.lookupService(serviceKind, serviceName, h.alloc.TaskGroup, req.TaskEnv)
 	if err != nil {
 		return err
 	}
@@ -179,7 +181,7 @@ func (h *envoyBootstrapHook) Prestart(ctx context.Context, req *ifs.TaskPrestart
 	// Envoy runs an administrative API on the loopback interface. There is no
 	// way to turn this feature off.
 	// https://github.com/envoyproxy/envoy/issues/1297
-	envoyAdminBind := buildEnvoyAdminBind(h.alloc, serviceName, req.Task.Name)
+	envoyAdminBind := buildEnvoyAdminBind(h.alloc, serviceName, req.Task.Name, req.TaskEnv)
 	resp.Env = map[string]string{
 		helper.CleanEnvVar(envoyAdminBindEnvPrefix+serviceName, '_'): envoyAdminBind,
 	}
@@ -271,12 +273,13 @@ func (h *envoyBootstrapHook) Prestart(ctx context.Context, req *ifs.TaskPrestart
 //
 // In host mode, use the port provided through the service definition, which can
 // be a port chosen by Nomad.
-func buildEnvoyAdminBind(alloc *structs.Allocation, serviceName, taskName string) string {
+func buildEnvoyAdminBind(alloc *structs.Allocation, serviceName, taskName string, taskEnv *taskenv.TaskEnv) string {
 	tg := alloc.Job.LookupTaskGroup(alloc.TaskGroup)
 	port := envoyBaseAdminPort
 	switch tg.Networks[0].Mode {
 	case "host":
-		for _, service := range tg.Services {
+		interpolatedServices := taskenv.InterpolateServices(taskEnv, tg.Services)
+		for _, service := range interpolatedServices {
 			if service.Name == serviceName {
 				mapping := tg.Networks.Port(service.PortLabel)
 				port = mapping.Value

